.github/workflows/root-ci.yml

```diff
@@ -131,6 +131,7 @@ jobs:
                     --repository     ${{ github.server_url }}/${{ github.repository }}"
 
       - name: Upload test results
+        if:   ${{ !cancelled() }}
```

Comment:
_Might_ need single quotes around `!cancelled()`, according to https://stackoverflow.com/a/58859404/17876693, although `${{ ... }}` is not used there. Let's see what happens

Reply:
This is copied from GitHub's documentation: https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions

Reply:
From stackoverflow:

> ``# Without quotes, `!cancelled()` is interpreted as a YAML tag``

Again that could just be wrong or `!` is evaluated differently inside an expression block.

Reply:
seems to work